### PR TITLE
new cluster option stonith-max-attempts 

### PR DESCRIPTION
**Pull request by aravind-kumar**
The maximum number of retry attempts for fencing to be made by crmd before giving up.Default value of 10 and accepts only positive non-zero values.
Added a new function check_positive_number() which return true if the input number is positive and false otherwise.

### CODE REVIEW

File: crmd/control.c [hunk 1]
```diff
@@ -952,6 +952,9 @@ pe_cluster_option crmd_opts[] = {
 	{ "stonith-watchdog-timeout", NULL, "time", NULL, NULL, &check_sbd_timeout,
 	  "How long to wait before we can assume nodes are safely down", NULL
         },
+        { "stonith-max-attempts",NULL,"integer",NULL,"10",&check_positive_number,
+          "The maximum number of stonith attempts before giving up."
```
Comment (kgaillot) [inline crmd/control.c]: Let's reword this as:

    How many times stonith can fail before it will no longer be attempted on a target

It's not much different, but it highlights that we will give up only for the target that failed, not all targets.

File: lib/common/utils.c [hunk 2]
```diff
@@ -162,6 +162,15 @@ check_number(const char *value)
 }
 
 gboolean
+check_positive_number(const char* value)
+{
+    if (check_number(value) && (value[0] != '-') && !(safe_str_eq(value,"0"))) {
```
Comment (kgaillot) [inline lib/common/utils.c]: This is a good test, but to be extra safe, I'm thinking it should be:

    if (safe_str_eq(value, INFINITY_S) || (crm_int_helper(value, NULL) > 0)) {

The difference is that crm_int_helper() allows whitespace before the number (because strtol() does). It's probably not an issue in practice, but it doesn't hurt to be on the safe side.

File: crmd/control.c (hunk 1 above)
Comment (aravind-kumar) [inline crmd/control.c]: Sure, I will reword the message 👍

File: lib/common/utils.c (hunk 2 above)
Comment (aravind-kumar) [inline lib/common/utils.c]: I agree this looks safe and better 👍

**Comment by kgaillot** [comment]
Thanks!
